src/event-log/event-stream.ts

```diff
@@ -0,0 +1,100 @@
+import type { MessageStore } from '../types/message-store.js';
+import type { EventsSubscribeMessage, EventsSubscription } from '../types/events-types.js';
+import type { EventStream, SubscriptionHandler } from '../types/subscriptions.js';
+import type { Filter, KeyValues } from '../types/query-types.js';
+import type { GenericMessage, GenericMessageSubscription } from '../types/message-types.js';
+
+import { EventEmitter } from 'events';
+import { EventsSubscribe } from '../interfaces/events-subscribe.js';
+import { EventsSubscriptionHandler } from '../handlers/events-subscribe.js';
+import { Message } from '../core/message.js';
+import { DwnError, DwnErrorCode } from '../core/dwn-error.js';
+
+const eventChannel = 'events';
+
+type EventStreamConfig = {
+  emitter?: EventEmitter;
+  reauthorizationTTL?: number;
+};
+
+export class EventStreamEmitter implements EventStream {
+  private eventEmitter: EventEmitter;
+  private reauthorizationTTL: number;
+
+  private isOpen: boolean = false;
+  private subscriptions: Map<string, SubscriptionHandler> = new Map();
+
+  constructor(config?: EventStreamConfig) {
+    this.reauthorizationTTL = config?.reauthorizationTTL || 0; // if set to zero it does not reauthorize
+
+    // we capture the rejections and currently just log the errors that are produced
+    this.eventEmitter = config?.emitter || new EventEmitter({ captureRejections: true });
+  }
+
+  private get eventChannel(): string {
+    return `${eventChannel}_bus`;
```

Comment:
This seems rather unnecessary, why add `_bus` and why not just a `const` or `private static readonly` and be done with it?